subprojects/signing/src/main/java/org/gradle/plugins/signing/SigningExtension.java

```diff
@@ -305,7 +308,47 @@ public void execute(Sign task) {
         return result;
     }
 
-    private Sign createSignTaskFor(final CharSequence name, Action<Sign> taskConfiguration) {
+    /**
+     * Creates signing tasks that sign {@linkplain PublicationInternal#getPublishableArtifacts()} all publishable artifacts} of the given publications.
+     *
+     * <p>The created tasks will be named "sign<i>&lt;publication name capitalized&gt;</i>Publication".
+     * That is, given a publication with the name "mavenJava" the created task will be named "signMavenJavaPublication".
+     *
+     * The signature artifacts for the created tasks are added to the {@linkplain PublicationInternal#getPublishableArtifacts()} publishable artifacts} of the given publications.
+     *
+     * @param publications The publications whose artifacts are to be signed
+     * @return the created tasks.
+     * @since 4.8
+     */
+    @Incubating
+    public List<Sign> sign(Publication... publications) {
+        List<Sign> result = new ArrayList<Sign>(publications.length);
+        for (final Publication publication : publications) {
+            if (publication instanceof PublicationInternal) {
```

Comment:
❌ You can do a direct cast here, as Publication is not to be implemented by user code. It should fail and not silently ignore the case where a user broke that contract.

Reply:
Done in 71838111765.